x/datadeal/keeper/deal_test.go

```diff
@@ -617,3 +617,71 @@ func (suite dealTestSuite) TestDataDeliveryVoteFaildInvalidStatus() {
 	err = suite.DataDealKeeper.VoteDataDelivery(ctx, dataDeliveryVote, signature)
 	suite.Require().ErrorIs(err, types.ErrDataDeliveryVote)
 }
+
+func (suite dealTestSuite) TestDeactivateDeal() {
```

Comment:
I suggest to write test with general scenario as below.
The purpose of this test is to check if the deal is inactive and its remained budget is sent to buyer back correctly.
But the below test code would not work with an error of the balance of deal is nil. 
Because the deal (dealID: 1) is set manually in `BeforeTest` using `keeper.SetDeal`, not using `keeper.CreateDeal`.
There is a logic for sending budget from buyer to deal in `keeper.CreateDeal`.

```go
func (suite dealTestSuite) TestDeactivateDeal() {
	ctx := suite.Ctx

	err := suite.FundAccount(ctx, suite.buyerAccAddr, suite.defaultFunds)
	suite.Require().NoError(err)

	getDeal, err := suite.DataDealKeeper.GetDeal(ctx, 1)
	suite.Require().NoError(err)

	dealAccAddr, err := sdk.AccAddressFromBech32(getDeal.Address)
	suite.Require().NoError(err)

	msgDeactivateDeal := &types.MsgDeactivateDeal{
		DealId:           1,
		RequesterAddress: suite.buyerAccAddr.String(),
	}

        // balance check before deactivation
	buyerBalance := suite.BankKeeper.GetBalance(ctx, suite.buyerAccAddr, assets.MicroMedDenom) // 10000 MED
	suite.Require().Equal(suite.defaultFunds, sdk.NewCoins(buyerBalance))

	dealBalance := suite.BankKeeper.GetBalance(ctx, dealAccAddr, assets.MicroMedDenom)
	suite.Require().Equal(sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(1000000000)), dealBalance) // 1000 MED

	err = suite.DataDealKeeper.DeactivateDeal(ctx, msgDeactivateDeal)
	suite.Require().NoError(err)

	getDeal, err = suite.DataDealKeeper.GetDeal(ctx, 1)
	suite.Require().NoError(err)

	suite.Require().Equal(getDeal.Status, types.DEAL_STATUS_INACTIVE)

        // balance check after deactivation. The remained budget should be sent to buyer correctly.
	buyerBalance = suite.BankKeeper.GetBalance(ctx, suite.buyerAccAddr, assets.MicroMedDenom)
	suite.Require().Equal(sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(11000000000)), buyerBalance)

	dealBalance = suite.BankKeeper.GetBalance(ctx, dealAccAddr, assets.MicroMedDenom)
	suite.Require().Equal(sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(0)), dealBalance) // 1000 MED

	//TODO: Check the DataVerification/DeliveryVote Queue are removed well
}
```

Reply:
Yes, when I tested there has an error of deal balance, so I put the following part:
```go
err = suite.BankKeeper.SendCoins(suite.Ctx, suite.buyerAccAddr, dealAccAddr, sdk.NewCoins(*getDeal.Budget))
suite.Require().NoError(err)
```
Is it okay to go like this?